### ISSUE TITLE
QVirtualScroll - unable to maintain focus on QInput after scrolling

### ISSUE THREAD

**Issue by nicholaszuccarelli** [quasarframework/quasar#8344]
I have provided a simple project here (also has a checkbox/radio which I used for the other bug I have reported for Cordova iOS).
https://github.com/nicholaszuccarelli/virtual_scroll_bug

In the case of this bug, if I have a QInput (or anything which can maintain focus) inside of a QVirtualScroll, it is not able to maintain its focus if the page scrolls / the virtual scroll container is changed.

This becomes a hassle on mobile because if the keyboard pops up, it almost always will remove focus from the input because the height of the page has updated.

I'm looking for a way to essentially maintain focus on the QVirtualScroll row until the row is no longer visible.

This is a problem on all platforms, although less obvious on desktop because the page doesn't change width/height for a keyboard, but is still an issue nonetheless.

**Comment by nicholaszuccarelli** [comment]
Update: it becomes a problem when the scroll-target=body

As displayed in this video, when I try to interact with one of the inputs at the bottom of the page, it will close the keyboard automatically, making it impossible to interact with that field 

https://user-images.githubusercontent.com/13761766/105126388-bbab7300-5b32-11eb-87e2-fb9ccec4d123.mp4

**Comment by pdanpdan** [comment]
@nicholaszuccarelli

Hello, can you please test if this fixes the problem for you?

In `package.json` replace `quasar` depencency with:
"quasar": "https://github.com/pdanpdan/quasar#quasar-pdan-v1.15.0-beta.2"

Then do a `yarn` or `npm install` and start the app again.

**Comment by nicholaszuccarelli** [comment]
On iOS, it seems to mostly work now. I did notice that sometimes when I would scroll, it would shift my focus from one QInput to another (unfortunately I couldn't replicate it on video). 

iOS video:
https://user-images.githubusercontent.com/13761766/105421303-0691cb80-5c96-11eb-88b1-4dee60724d19.MP4

Android is very broken. It will sometimes loose focus on the element when I scroll, and when I am at the bottom of the page I still cannot click the very bottom elements.

**Comment by pdanpdan** [comment]
You must use key on the items in the list, else vue reuses the components and you'll end up waiting something else. Use index provided in scope of some unique Id in your data.

**Comment by nicholaszuccarelli** [comment]
> You must use key on the items in the list, else vue reuses the components and you'll end up waiting something else. Use index provided in scope of some unique Id in your data.

As per the same project (this page specifically), how would I apply a key given that it uses 
`<template v-slot="{ item }">`
https://github.com/nicholaszuccarelli/virtual_scroll_bug/blob/master/src/pages/Index.vue

**Comment by pdanpdan** [comment]
```vue
      <template v-slot="{ item, index }">
        <q-item :key="index">
          <q-item-section>
            <q-input
              :value="item"
            />
          </q-item-section>
        </q-item>
      </template>
```

**Comment by nicholaszuccarelli** [comment]
Ah right!
I'll give this a shot and see if it works now.

**Comment by nicholaszuccarelli** [comment]
@pdanpdan Works perfectly on iOS, Android and Chrome now after adding an index.

**Comment by nicholaszuccarelli** [comment]
I tested beta 2 on my production app as well and it fixes the issues I was experiencing on there too.

**Comment by rstoenescu** [comment]
Fix will be available in Quasar v1.15.1
